Task1_Ivan/PtrStringList.cpp

```diff
@@ -0,0 +1,249 @@
+#include "PtrStringList.h"
+
+namespace slst
+{
+
+    void slst::StringListInit(char*** list)
+    {
+        list[0] = StringListAllocateNode();
+    }
+
+    void slst::StringListDestroy(char*** list)
+    {
+        if (list == nullptr || list[0] == nullptr)
+            return;
+        char** currentNode = list[0];
+        while (currentNode /* != LIST_END */)
+        {
+            char** next = StringListGetNext(currentNode);
+            free(currentNode[0]);
+            free(currentNode);
+            currentNode = next;
+        }
+        list[0] = nullptr;                                      // Terminate end of the list
+    }
+
+    void slst::StringListAdd(char** list, String str)
+    {  
+        char** newNode = nullptr;
+        if (list && str) // If list is not empty then append
+        {
+            newNode = StringListAllocateNode();
+            StringListAssignVal(newNode, str);
+            char** currentNode = list;
+            char** next;
+            while (next = StringListGetNext(currentNode))       // Find end of the list
+                currentNode = next;
+            StringListSetNext(currentNode, newNode);            // Append new node
+        }
+        //else throw...                                       
+    }
+
+    void slst::StringListRemove(char** head, const String str)
+    {

```

Comment:
Needs to add verification for input parameters.

Reply:
Since there is no dereferencing before loops, i use while() to check for null, including passed parameters. If that is not okay, i will add additional check at the start.

Reply:
Yes, i see. Will add checks for head or str being null